### PR TITLE
Added SPI1 object for Teensy 3.5 test board

### PR DESCRIPTION
**Pull request by KurtE**
These changes plus some changes in core (added register defines, plus
added logical SPCR1 register), allowed me to do quick hack to
serialFlash library and test and initialize flash memory using 3.5 test
board connected to prop shield using pins 0, 1, 20 and 6

### CODE REVIEW

File: SPI.cpp [hunk 1]
```diff
@@ -308,6 +309,132 @@ uint8_t SPIClass::setCS(uint8_t pin)
...
+			SPI1_CTAR0 = ctar;
+			SPI1_CTAR1 = ctar | SPI_CTAR_FMSZ(8);
+			SPI1_MCR = mcr;
+		}
+	}
+}
+
+void SPI1Class::setBitOrder(uint8_t bitOrder)
+{
+	SIM_SCGC6 |= SIM_SCGC6_SPI0;
```
Comment (sumotoy) [inline SPI.cpp]: should be SIM_SCGC6 |= SIM_SCGC6_SPI1; ?

File: SPI.cpp [hunk 2]
```diff
@@ -308,6 +309,132 @@ uint8_t SPIClass::setCS(uint8_t pin)
...
+		ctar |= SPI_CTAR_LSBFE;
+	} else {
+		ctar &= ~SPI_CTAR_LSBFE;
+	}
+	updateCTAR1(ctar);
+}
+
+void SPI1Class::setDataMode(uint8_t dataMode)
+{
+	SIM_SCGC6 |= SIM_SCGC6_SPI0;
```
Comment (sumotoy) [inline SPI.cpp]: SIM_SCGC6 |= SIM_SCGC6_SPI1;

File: SPI.cpp [hunk 3]
```diff
@@ -308,6 +309,132 @@ uint8_t SPIClass::setCS(uint8_t pin)
...
+
+	// TODO: implement with native code
+
+	SPCR = (SPCR & ~SPI_MODE_MASK) | dataMode;
+}
+
+void SPI1Class::setClockDivider_noInline(uint32_t clk)
+{
+	SIM_SCGC6 |= SIM_SCGC6_SPI0;
```
Comment (sumotoy) [inline SPI.cpp]: SIM_SCGC6 |= SIM_SCGC6_SPI1;

**Comment by KurtE** [comment]
Updated misses to SIM_SCGC6

**Comment by PaulStoffregen** [comment]
Sorry about the delay on merging this and others.  I've been reluctantly debating when to cut off stuff for 1.29... and really, I need to get a stable release done for Teensy 3.2.  Soon we'll begin 1.30 betas in support of the upcoming boards.

**Comment by KurtE** [comment]
Sounds reasonable.  I am still playing with it as Sumotoy mentioned there is still some issues using SPCR1 logical register, which I am playing with.

**Pull request merged**